pkg/networkmanager/dialogs-common.jsx

```diff
@@ -167,28 +169,9 @@ export const NetworkModal = ({ dialogError, help, idPrefix, title, onSubmit, chi
 
 export const NetworkAction = ({ buttonText, iface, connectionSettings, type }) => {
     const Dialogs = useDialogs();
-    const [showAddTeam, setShowAddTeam] = useState(undefined);
     const model = useContext(ModelContext);
 
-    useEffect(() => {
-        if (type != "team")
-            return;
-
-        /* HACK - hide "Add team" if it doesn't work due to missing bits
-         * https://bugzilla.redhat.com/show_bug.cgi?id=1375967
-         * We need both the plugin and teamd
-         */
-        cockpit.script("test -f /usr/bin/teamd && " +
-                       "( test -f /usr/lib*/NetworkManager/libnm-device-plugin-team.so || " +
-                       "  test -f /usr/lib*/NetworkManager/*/libnm-device-plugin-team.so || " +
-                       "  test -f /usr/lib/*-linux-gnu/NetworkManager/libnm-device-plugin-team.so || " +
-                       "  test -f /usr/lib/*-linux-gnu/NetworkManager/*/libnm-device-plugin-team.so)",
-                       { err: "ignore" })
-                .then(() => setShowAddTeam(true))
-                .fail(() => setShowAddTeam(false));
-    }, [type]);
-
-    if (type == "team" && !showAddTeam)
+    if (!model.get_manager().Capabilities || (type == "team" && !model.get_manager().Capabilities.includes(NM_CAPABILITY_TEAM)))
```

Comment:
Hmm, I would expect Capabilites to always have a value when this component is rendered... (`model.ready` should be true). I try to figure this out.

Reply:
Aha, this happens when NetworkManager disappears and then comes back while the page is open.  I guess all the components survive this because the API wrapper specifies safe defaults for properties, so let's do this as well for Capabilities:

```
diff --git a/pkg/networkmanager/dialogs-common.jsx b/pkg/networkmanager/dialogs-common.jsx
index ba8a59034b..5f946aeaaa 100644
--- a/pkg/networkmanager/dialogs-common.jsx
+++ b/pkg/networkmanager/dialogs-common.jsx
@@ -171,7 +171,7 @@ export const NetworkAction = ({ buttonText, iface, connectionSettings, type }) =
     const Dialogs = useDialogs();
     const model = useContext(ModelContext);
 
-    if (!model.get_manager().Capabilities || (type == "team" && !model.get_manager().Capabilities.includes(NM_CAPABILITY_TEAM)))
+    if (type == "team" && !model.get_manager().Capabilities.includes(NM_CAPABILITY_TEAM))
         return null;
 
     const con = iface && iface.MainConnection;
diff --git a/pkg/networkmanager/interfaces.js b/pkg/networkmanager/interfaces.js
index 3c5760fc07..8e64e4ade8 100644
--- a/pkg/networkmanager/interfaces.js
+++ b/pkg/networkmanager/interfaces.js
@@ -1155,7 +1155,7 @@ export function NetworkManagerModel() {
         ],
 
         props: {
-            Capabilities: { },
+            Capabilities: { def: [] },
             Version: { },
             Devices: {
                 conv: conv_Array(conv_Object(type_Device)),
```